### ISSUE TITLE
Errors dispatched by middlewares are not taken into account

### ISSUE THREAD

**Issue by julien51** [unlock-protocol/unlock#1977]
In the middleware we dispatch errors sent by the web3Service and/or walletService, such as `NOT_ENABLED_IN_PROVIDER` for example.
At this point, these are "ignored" by the application... rather than being taken into account by the `GlobalErrorProvider`.

**Comment by cellog** [comment]
The global error provider was never designed for this purpose. It was designed to create errors, by checking conditions and passing them down through a separate context to be consumed by GlobalErrorConsumer. It is entirely separate from redux. Thus errors set in redux should be handled by components in the normal way, with other components.

Perhaps it is time to refactor error handling? I think the current design is confusing and a code smell now, even though when I wrote it in December it seemed appropriate to the codebase then, the fact that I needed to explain this design is already a code smell!

One possible temporary solution is to refactor the `GlobalErrorConsumer` to also listen for errors in redux? At least until we can talk about what (if any) larger architecture changes are needed to make the error handling easier to manage. I have not had time or energy to think about this, so these are mostly brain dump thoughts to get the discussion going
